pkg/platform/model/buildplanner.go

```diff
@@ -102,28 +103,36 @@ func NewBuildPlannerModel(auth *authentication.Auth) *BuildPlanner {
 }
 
 func (bp *BuildPlanner) FetchBuildResult(commitID strfmt.UUID, owner, project string) (*BuildResult, error) {
-	resp := &bpModel.BuildPlan{}
-	err := bp.client.Run(request.BuildPlan(owner, project, commitID.String()), resp)
+	logging.Debug("FetchBuildResult")
+	logging.Debug("CommitID: %s", commitID.String())
+	logging.Debug("Owner: %s", owner)
+	logging.Debug("Project: %s", project)
+	resp := bpModel.NewBuildPlanResponse(owner, project)
+	err := bp.client.Run(request.BuildPlan(commitID.String(), owner, project), resp)
 	if err != nil {
 		return nil, errs.Wrap(err, "failed to fetch build plan")
 	}
 
-	// Check for errors in the response
-	if resp.Project.Type == bpModel.NotFound {
-		return nil, locale.NewError("err_buildplanner_project_not_found", "Build plan does not contain project")
+	data, err := json.MarshalIndent(resp, "", "  ")
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not unmarshal JSON")
 	}
-	if resp.Project.Commit.Type == bpModel.NotFound {
-		return nil, locale.NewError("err_buildplanner_commit_not_found", "Build plan does not contain commit")
+	logging.Debug("Response:", string(data))
+
+	build, err := resp.Build()
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not get build from response")
 	}
-	if resp.Project.Commit.Build.Type == bpModel.BuildResultPlanningError {
+
+	if build.PlanningError != nil {
 		var errs []string
 		var isTransient bool
-		for _, se := range resp.Project.Commit.Build.SubErrors {
+		for _, se := range build.SubErrors {
```

Comment:
This feels like something that should be handled in the `Build()` function (ie. the error contained could already address this use-case).